torchvision/datapoints/_dataset_wrapper.py

```diff
@@ -288,30 +290,40 @@ def wrapper(idx, sample):
         if not target:
             return image, dict(image_id=image_id)
 
+        spatial_size = tuple(F.get_spatial_size(image))
         batched_target = list_of_dicts_to_dict_of_lists(target)
+        target = {}
 
-        batched_target["image_id"] = image_id
+        if "image_id" in target_keys:
+            target["image_id"] = image_id
```

Comment:
Instead of adding all those `if` blocks, should we just keep these parts intact and just drop the unnecessary keys at the end?

Reply:
Somewhat a trade-off. If we leave everything intact until the very end, we pay for the in between, e.g. decode masks just to throw them away shortly after. I don't have a good solution for this yet. It would also need to factor in the `target_types="all"` case.